warp/aggregator/aggregator_test.go

```diff
@@ -390,9 +477,10 @@ func TestAggregateSignatures(t *testing.T) {
 			ctrl := gomock.NewController(t)
 			require := require.New(t)
 
-			a := tt.aggregatorFunc(ctrl)
+			ctx, cancel := tt.contextWithCancelFunc()
+			a := tt.aggregatorFunc(ctrl, cancel)
```

Comment:
should we add a check like:
`if cancel != nil { defer cancel() }` to avoid leaking goroutine or is it guaranteed that `aggregatorFunc` calls cancel?

Reply:
Currently in all the places where cancel is not nil, aggregatorFunc calls `cancel`. 
I can put this check in place for safety